Generate test artifacts

rosstimothy opened the pull request:
Updates test targets to output json and junit files. These files will be forwarded to a durable storage location in the future to be consumed for analytical purposes.

fheinecke reviewed (dismissed):
In an effort to reduce our dependency on GHA, we're trying to avoid relying on GHA-specific features, such as artifact storage. Can you update this to push to a S3 bucket or otherwise make these available without using GHA artifacts?

fheinecke commented:
Completely get that we're currently doing this elsewhere. The goal is to stop doing this going forward for all new changes (not just new workflows, but changes to existing workflows as well). Longer term, we're working to move existing GHA-specific dependencies away from GHA. Adding additional logic that depend on GHA-specific features moves us backwards, so we really want to avoid this.

If you need assistance setting up required infra (S3, IAM role) for this, let me know and I can point you to some resources we have available.

okraport reviewed (approved):
> Longer term, we're working to move existing GHA-specific dependencies away from GHA. 

Agreed, that is the desired end state of this work as I understand it so I think we are all in alignment on that. And I completely understand the hesitation to add this change now. 

> Adding additional logic that depend on GHA-specific features moves us backwards, so we really want to avoid this.

That said, does this change in particular carry risk of any further issues / instabilities? The intention here is to start simple and iterate and as I understand it this is the simplest thing we can do right now to start querying this data.

rosstimothy committed:
Generate test artifacts

Updates test targets to output json and junit files. These files
will be forwarded to a durable storage location in the future to
be consumed for analytical purposes.
